cpp/tests/cppflow_model.cc

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <vector>
+#include <catch.hpp>
+
+#include <sopt/logging.h>
+#include <sopt/types.h>
+#include <sopt/utilities.h>
+#include <sopt/cppflow_utils.h>
+#include <cppflow/cppflow.h>
+#include "cppflow/ops.h"
+#include "cppflow/model.h"
+
+// This header is not part of the installed sopt interface
+// It is only present in tests
+#include <tools_for_tests/directories.h>
+#include <tools_for_tests/tiffwrappers.h>
+
+// \min_{x} ||\Psi^Tx||_1 \quad \mbox{s.t.} \quad ||y - Ax||_2 < \epsilon and x \geq 0
+
+typedef double Scalar;
+typedef sopt::Image<Scalar> Image;
+
+
+TEST_CASE("Cppflow Model"){
+
+  // read in image
+  std::string const input_image = "cameraman256";
+  Image const image = sopt::notinstalled::read_standard_tiff(input_image);
+
+  int const image_rows = image.rows();
+  int const image_cols = image.cols();
+
+  cppflow::tensor input_tensor = sopt::cppflowutils::convert_image_to_tensor(image, image_rows, image_cols);
+
+  // Read in model
+  cppflow::model model(std::string(sopt::notinstalled::models_directory() + "/DnCNN/snr_15_model.pb/"));
+
+  // Run model on image
+  auto output_vector = model({{"serving_default_input0:0", input_tensor}}, {"StatefulPartitionedCall:0"});
```

Comment:
Just putting a note here as a reminder, we can do this later.

```suggestion
  // Run model on image
  // TODO: Automatically detect the string parameters, see issue #320 
  auto output_vector = model({{"serving_default_input0:0", input_tensor}}, {"StatefulPartitionedCall:0"});
```